HomeWork/src/main/java/streams/ThreadSafeList.java

```diff
@@ -0,0 +1,61 @@
+package streams;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+public class ThreadSafeList<T> {
+    private final Object lock = new Object();
+    private Object[] array;
+
+    public Object[] getArray() {
+        return array;
+    }
+
+    public void setArray(Object[] array) {
+        this.array = array;
+    }
+
+    public ThreadSafeList() {
+        setArray(new Object[0]);
+    }
+
+    private int size() {
+        return getArray().length;
+    }
+
+    public void add(T element) {
+        synchronized (lock) {
+            Object[] ne = getArray();
+            int length = size();
+            ne = Arrays.copyOf(ne, length + 1);
```

Comment:
1. It's not efficient to copy the whole array every time you add an element. Copying only should be done only if the current array is full.
2. Add operation should be like this: `elements[index++] = element;`